File: python/Amenity.py
from abc import ABC, abstractmethod

class Amenity(ABC):
    @property
    def price(self):
        return self.__price
    @price.setter
    def price(self, price):
        self.__price = price
    
    def __init__(self):
        self.__price = 0

    @abstractmethod
    def getDescription(self):
        pass

File: Amenity.java
public abstract class Amenity {
    private float price;

    public Amenity() {}

    public Amenity(float price) {
        this.price = price;
    }

    public float getPrice() {
        return this.price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public abstract String getDescription();

}